net/src/main/java/io/bitcoinsv/jcl/net/protocol/handlers/block/BlockDownloaderHandlerImpl.java

```diff
@@ -515,6 +527,45 @@ private boolean isDownloadComplete(String blockHash) {
         return result;
     }
 
+    private String getHashFromMsg(BitcoinMsg<?> msg) {
+        String blockHash = (msg.is(BlockMsg.MESSAGE_TYPE))
+                ? Utils.HEX.encode(((BlockMsg)msg.getBody()).getBlockHeader().getHash().getBytes())
+                : (msg.is(RawBlockMsg.MESSAGE_TYPE))
+                ? Utils.HEX.encode(((RawBlockMsg) msg.getBody()).getBlockHeader().getHash().getBytes())
```

Comment:
Possible ClassCastException, BlockMsg and RawBlockMsg have identical MESSAGE_TYPE values (in current version).